src/wdl_aid/wdl_aid.py

```diff
@@ -244,11 +244,67 @@ def gather_inputs(workflow: WDL.Workflow
     return inputs, required_inputs
 
 
+def gather_entries(name_and_bindings: List[Tuple[str, WDL.Env.Binding]],
+                   parameter_meta: Dict[str, Any], category_key: str,
+                   fallback_category: str, description_key: str,
+                   fallback_description: str,
+                   fallback_description_to_object: bool,
+                   excluded_names: List[str], required_names: List[str] = []
+                   ) -> Tuple[Dict[str, List[Dict[str, Any]]], List[str]]:
+    """
+    :param name_and_bindings: A tuple in which the first element is the
+    fully qualified name of the binding in the second element.
+    :param parameter_meta: A dictionary containing the parameter_meta
+    information.
+    :param category_key: The key used in parameter_meta for categories.
+    :param fallback_category: The default category.
+    :param description_key: The key used in parameter_meta for
+    descriptions.
+    :param fallback_description: The default description.
+    :param fallback_description_to_object: Whether or not the entire
+    object should be returned for a given object if the description
+    key is not found.
+    :param excluded_names: The names to exclude.
+    :param required_names: The names to separate into a "required" category.
+    :return: The entries and a list of names for which parameter_meta is
+    missing.
+    """
+    entries = {}
+    missing_parameter_meta = []
+    for name, binding in name_and_bindings:
+        if name in excluded_names:
+            continue
+        if name not in parameter_meta:
+            missing_parameter_meta.append(name)
+        category = ("required" if name in required_names
+                    else get_category(parameter_meta, name, category_key,
+                                      fallback_category))
+        entry_type = (str(binding.value.type)
+                      if hasattr(binding.value, "type")
+                      else str(binding.value))
+        entry = {
+            "name": name,
+            "type": entry_type,
+            "description":
+                get_description(parameter_meta, name, description_key,
+                                fallback_description,
+                                fallback_description_to_object)
+        }
+        if hasattr(binding.value, "expr"):
+            entry["default"] = (str(binding.value.expr)
+                                if binding.value.expr is not None else None)
+        try:
+            entries[category].append(entry)
+        except KeyError:
+            entries[category] = [entry]
```

Comment:
Can be replaced with `entries[category].append(entry)` entirely if entries is a defaultdict. Of course this comes down to preference.